pgx/src/datum/sql_entity_graph/extension_sql.rs

```diff
@@ -167,79 +145,34 @@ impl Display for SqlDeclaredEntity {
 impl SqlDeclaredEntity {
     pub fn build(variant: impl AsRef<str>, name: impl AsRef<str>) -> eyre::Result<Self> {
         let name = name.as_ref();
+        let data = SqlDeclaredEntityData {
+            sql: name
+                .split("::")
+                .last()
+                .ok_or_else(|| eyre::eyre!("Did not get SQL for `{}`", name))?
+                .to_string(),
+            name: name.to_string(),
+            option: format!("Option<{}>", name),
+            vec: format!("Vec<{}>", name),
+            vec_option: format!("Vec<Option<{}>>", name),
+            option_vec: format!("Option<Vec<{}>>", name),
+            option_vec_option: format!("Option<Vec<Option<{}>>", name),
+            array: format!("Array<{}>", name),
+            option_array: format!("Option<{}>", name),
+            varlena: format!("Varlena<{}>", name),
+            pg_box: vec![
+                format!("pgx::pgbox::PgBox<{}>", name),
+                format!("pgx::pgbox::PgBox<{}, pgx::pgbox::AllocatedByRust>", name),
+                format!(
+                    "pgx::pgbox::PgBox<{}, pgx::pgbox::AllocatedByPostgres>",
+                    name
```

Comment:
Suggest using `std::any::type_name` rather than hard-coding these strings. Will become brittle if things ever are refactored.

Reply:
we can't here as the name itself is dynamically generated.  :(  But you're right -- we'll get tripped up by this at some point.  We do have tests that use `PgBox` so they'll at least light up when we move things around.